### PR TITLE
Implement #182 - Load GTFS entities into memory

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR is a first draft that allows to load GTFS entities into memory. 
For now, only files `agency.txt` and `routes.txt` are handled. In the near future, all files should be supported and a use case to exclude files from validation will be implemented. 

**Expected behavior:** 

Rows of files `agency.txt` and `routes.txt` should be internally represented as Route and Agency objects. 

Run the following config in debug mode (of IntelliJ) and observe that GtfsDataRepository has been filled with routes and agency objects: `-u https://openmobilitydata.org/p/societe-de-transport-de-montreal/39/latest/download -z input.zip -e input -p -o output`

**routes.txt** contains 231 row, expecting the same number of Route entities: OK

Capture d’écran, le 2020-05-05 à 13 52 16

**agency.txt** contains 1 row, expecting the same number of Agency entities: OK
Capture d’écran, le 2020-05-05 à 13 53 12

Capture d’écran, le 2020-05-05 à 13 53 54

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: application/cli-app/src/main/java/org/mobilitydata/gtfsvalidator/Main.java [hunk 1]
```diff
@@ -54,14 +57,30 @@ public static void main(String[] args) {
...
+                // base validation + load gtfs entities into memory
                 filenameList.forEach(filename -> {
                     config.validateHeadersForFile(filename).execute();
                     config.validateAllRowLengthForFile(filename).execute();
 
                     ParseSingleRowForFile parseSingleRowForFile = config.parseSingleRowForFile(filename);
                     while (parseSingleRowForFile.hasNext()) {
-                        config.validateGtfsTypes().execute(parseSingleRowForFile.execute());
+                        final ParsedEntity parsedEntity = parseSingleRowForFile.execute();
+                        config.validateGtfsTypes().execute(parsedEntity);
```
Comment (fabrice-v) [inline application/cli-app/src/main/java/org/mobilitydata/gtfsvalidator/Main.java:73]: Fix existing issue: retrieve ValidateGtfsTypes use case instance before the while loop and call execute on it once per parsedEntity

File: application/cli-app/src/main/java/org/mobilitydata/gtfsvalidator/Main.java [hunk 2]
```diff
@@ -54,14 +57,30 @@ public static void main(String[] args) {
...
+
+                        // load gtfs entities into memory
+                        // in the future all filename in filenameList will be processed. For now focusing on routes.txt
+                        // and agency.txt.
+                        // filenames in filenameList will be determined using a dependency tree defined by a JSON file,
+                        // and command lines or configuration file will be used to exclude files from the validation
+                        // process.
+                        if (toLoadIntoMemory.contains(filename)) {
+                            config.loadAllEntityIntoMemory().execute(parsedEntity);
```
Comment (fabrice-v) [inline application/cli-app/src/main/java/org/mobilitydata/gtfsvalidator/Main.java:82]: Retrieve use case instance before while loop and call execute on it with each parsed entity

File: usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/LoadAllEntityIntoMemory.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.ParsedEntity;
+
+/**
+ * Use case to load into memory GTFS entities. This use case is called after parsing a row from a GTFS .txt file.
+ */
+public class LoadAllEntityIntoMemory {
```
Comment (fabrice-v) [inline usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/LoadAllEntityIntoMemory.java:24]: This use case is a switch container. After discussion, we decided to implement the switch directly in main and remove this class.

File: config/src/main/java/org/mobilitydata/gtfsvalidator/config/DefaultConfig.java [hunk 4]
```diff
@@ -143,4 +141,12 @@ public LogExecutionInfo logExecutionInfo() {
     public PrintHelp printHelp() {
         return new PrintHelp(execParamRepo, new HelpFormatter());
     }
+
+    public LoadAllEntityIntoMemory loadAllEntityIntoMemory() {
```
Comment (fabrice-v) [inline config/src/main/java/org/mobilitydata/gtfsvalidator/config/DefaultConfig.java:145]: Replace with a method per use case.

File: config/src/main/java/org/mobilitydata/gtfsvalidator/config/DefaultConfig.java [hunk 5]
```diff
@@ -143,4 +141,12 @@ public LogExecutionInfo logExecutionInfo() {
     public PrintHelp printHelp() {
         return new PrintHelp(execParamRepo, new HelpFormatter());
     }
+
+    public ProcessParsedAgency processParsedAgency() {
+        return new ProcessParsedAgency(resultRepo, gtfsDataRepository, new Agency.AgencyBuilder(new ArrayList<>()));
```
Comment (fabrice-v) [inline config/src/main/java/org/mobilitydata/gtfsvalidator/config/DefaultConfig.java:146]: After discussion, we decided to remove parametric constructors from our builder classes

File: application/cli-app/src/main/java/org/mobilitydata/gtfsvalidator/Main.java [hunk 6]
```diff
@@ -54,22 +59,50 @@ public static void main(String[] args) {
...
+                        validateGtfsTypes.execute(parsedEntity);
+
+                        // load gtfs entities into memory
+                        // in the future all filename in filenameList will be processed. For now focusing on routes.txt
+                        // and agency.txt.
+                        // filenames in filenameList will be determined using a dependency tree defined by a JSON file,
+                        // and command lines or configuration file will be used to exclude files from the validation
+                        // process.
+                        if (toLoadIntoMemory.contains(filename)) {
+                            switch (parsedEntity.getRawFileInfo().getFilename()) {
```
Comment (fabrice-v) [inline application/cli-app/src/main/java/org/mobilitydata/gtfsvalidator/Main.java:89]: In that context we can use `filename` directly

File: config/src/main/java/org/mobilitydata/gtfsvalidator/config/DefaultConfig.java (hunk 5 above)
Comment (lionel-nj) [reply config/src/main/java/org/mobilitydata/gtfsvalidator/config/DefaultConfig.java:146]: See [related PR](https://github.com/MobilityData/gtfs-validator/pull/187)

**Pull request merged**